libpod/container_internal_linux.go

```diff
@@ -778,6 +778,22 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		}
 	}
 
+	// Pass down the LISTEN_* environment (see #10443).
+	for _, key := range []string{"LISTEN_PID", "LISTEN_FDS", "LISTEN_FDNAMES"} {
+		if val, ok := os.LookupEnv(key); ok {
+			if key == "LISTEN_PID" {
+				// The primary process within the container
+				// will be PID=1, so this value needs to be
+				// reset.
+				//
+				// NOTE: runc does the same but crun doesn't,
+				// so let's make sure it's always set.
+				val = "1"
+			}
+			g.AddProcessEnv(key, val)
```

Comment:
should we do this only when `--pid=private`?

Reply:
Oh, that is a very fair point. Thanks for catching!

Reply:
I'll experiment a bit.  At least runc would override it unconditionally if LISTEN_FDS are set.